```{r setup, echo=FALSE, message=FALSE, include=FALSE, error=FALSE}

library(cowplot)
library(dplyr)
library(tidyr)
library(Gviz)
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
library(tximport)
library(DESeq2)
library(magrittr)
library(pander)

# Output folder for this document
options(knitr.figure_dir = "main_analysis")

source("shared/knitr_common.r")
source("shared/granges_common.r")
source("shared/metapeaks.r")

source("shared/promoter_element_enrichments.r")
panderOptions('knitr.auto.asis', TRUE)

source("shared/flybase.r")
source("shared/gostats.r")

cores <- function() { return(16) }

knitr::opts_chunk$set(dev=c("png", "pdf"))

library(cowplot)

```


```{r header_child, child="child_docs/header_child.Rmd"}
```

## Overview

  - Define late induced genes
  - GO analysis of late genes
  - Seperate late genes to groups by pol II occupancy in different tissues
  - RNA-seq expression
  - Promoter element enrichments
  - ImaGO terms
  - Gene length
  - ATAC 14-17h
  - MNase 2-4h and 14-17hrs

## Late induced genes and housekeeping genes

Whole-embryo wild-type 2-4h (Nina) and 14-17h (Malini) were processed with `kallisto` and compared with `DESeq2`. 

Late-induced genes are those considered up-regulated late versus early with a TPM of less than 25 early and greater than 100 late.

As a comparison group, we will also define housekeeping genes as those that show no differential expression between the early and late time points and are expressed with an TPM of at least 100 in both stages.


```{r define_late_and_housekeeping_genes, message=FALSE}
all_fb_genes <- flybase_txs_granges()$fb_gene_id %>% unique


files <- file.path("../fastq/rnaseq/kallisto/", list.files("../fastq/rnaseq/kallisto/"), "abundance.h5")
names(files) <- list.files("../fastq/rnaseq/kallisto/")
files %<>% {.[names(.)%in% c("wt_2to4h_1","wt_2to4h_2","wt_2to4h_3","wt_14to17h_1","wt_14to17h_2","wt_14to17h_3","wt_14to17h_4rr")]}


txi <- tximport(files, type = "kallisto", txOut = TRUE)


sampleTable <- data.frame(condition = factor(rep(c("wt_14to17h", "wt_2to4h"), c(4,3))))
rownames(sampleTable) <- colnames(txi$counts)
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~condition)

dds<-DESeq(dds)



rnaseq <- results(dds,alpha = 0.01) %>% as.data.frame %>% dplyr::mutate(fb_gene_id = row.names(.))

txi$abundance %>% as.data.frame %>% dplyr::mutate(fb_gene_id = row.names(.), wt_14to17h = (wt_14to17h_1+wt_14to17h_2+wt_14to17h_3+wt_14to17h_4rr)/4, wt_2to4h = (wt_2to4h_1+wt_2to4h_2+wt_2to4h_3)/3) %>% dplyr::select(fb_gene_id,wt_2to4h,wt_14to17h) ->TPM

rnaseq %<>% merge(TPM)

late_genes <- rnaseq %>% 
              filter(wt_14to17h > 10 & wt_2to4h < 2 & padj < 0.01) %>%
              dplyr::mutate(gene_group = "Late-induced genes")

hk_genes <- rnaseq %>%
            filter(wt_14to17h > 10 & wt_2to4h > 10 & padj > 0.01) %>%
            dplyr::mutate(gene_group = "Housekeeping genes")

bind_rows(late_genes, hk_genes) %>%
  group_by(gene_group) %>%
  dplyr::summarize(count = n(), 
            median_TPM = median(wt_14to17h)) %>%
  pander(caption="Summary of late-induced genes")


lst<-list(late_genes=late_genes,hk_genes = hk_genes)

rnaseq %>% dplyr::select(fb_gene_id,log2FoldChange,pvalue,padj,wt_2to4h,wt_14to17h) %>% write.csv(figure_path("Dataset_EV1_whole_embryo_14to17hr_expression.csv"))

```


```{r}

all_fb_genes <- flybase_txs_granges()$fb_gene_id %>% unique
files <- file.path("../fastq/rnaseq/kallisto/", list.files("../fastq/rnaseq/kallisto/"), "abundance.h5")
names(files) <- list.files("../fastq/rnaseq/kallisto/")
files %<>% {.[names(.)%in% c("wt_14to14.5h_1","wt_14to14.5h_2","wt_17to17.5h_1","wt_17to17.5h_2")]}
files

txi <- tximport(files, type = "kallisto", txOut = TRUE)

sampleTable <- data.frame(condition = factor(rep(c("wt_14to14.5h", "wt_17to17.5h"), c(2,2))))
rownames(sampleTable) <- colnames(txi$counts)


txi$abundance %>% as.data.frame %>% dplyr::mutate(fb_gene_id = row.names(.), wt_14to14.5h = (wt_14to14.5h_1+wt_14to14.5h_2)/2, wt_17to17.5h = (wt_17to17.5h_1+wt_17to17.5h_2)/2) %>% dplyr::select(fb_gene_id,wt_14to14.5h,wt_17to17.5h) ->TPM

rnaseq %>% merge(TPM) -> rnaseq2

rnaseq2 %>% dplyr::select(fb_gene_id,log2FoldChange,pvalue,padj,wt_2to4h,wt_14to17h) %>% write.csv(figure_path("Dataset_EV2_whole_embryo_14and17hr_expression.csv"))

```
## GO analysis

Below are the top 5 GO categories for each gene group (excluding categories that contain 5 or less genes). The full results are in the output folder for this document as `go_results.xls`.

```{r go_analysis, results="asis"}
go_term_analysis<-function(){ panderOptions('knitr.auto.asis', FALSE)

gene_groups <- list("Late-induced genes"=late_genes$fb_gene_id,
                    "Housekeeping genes"=hk_genes$fb_gene_id)

gene_universe <- rnaseq$fb_gene_id


go_results<-gene_groups %>%
    lapply(run_go, gene_universe) %>%
    bind_rows(.id="gene_group")


for(i in unique(go_results$gene_group)) {
  filter(go_results, gene_group == i & Size >= 5) %>%
    arrange(Pvalue) %>%
    head(5) %>%
    dplyr::select(-gene_group, -OddsRatio, -ExpCount) %>%
    pander(caption=paste0("Gene group: ", i))
}

write.csv(go_results, file=figure_path("go_results.csv"))
}

if(!file.exists(figure_path("go_results.csv"))){go_term_analysis()}

#Late genes go terms figure based on excel sheet

read.csv(figure_path("go_results.csv")) ->dfs

dfs %<>% filter(Term %in% c("locomotory behavior","proteolysis","muscle system process","anterograde trans-synaptic signaling","amino sugar metabolic process","chitin-based cuticle development")) 

dfs$Term %<>% factor(c("locomotory behavior","proteolysis","muscle system process","anterograde trans-synaptic signaling","amino sugar metabolic process","chitin-based cuticle development"))

dfs$Pvalue %<>% {log10(.) *(-1)}

dfs %>% ggplot(aes(Term,Pvalue))+geom_bar(stat="identity") + coord_flip()
```

## Dividing genes by Pol II presence in different tissues

Next, we will calculate Pol II enrichment levels at the promoters of all genes and divide the late-induced genes into groups based on the number of tissues where Pol II is detected at least 2-fold above background. We will use the following Pol II samples:

```{r polii_samples, message=FALSE}

library(Biostrings); library(BSgenome.Dmelanogaster.UCSC.dm6)
library(reshape2); library(ggseqlogo); library(gridExtra)


calculate_polii_enrichment<-function(){
  
pol_samples_df <- read.csv("data/samples.csv") %>%
                  filter(factor == "polii") %>%
                  dplyr::select(-starts_with("window_"))

pol_samples_df %>% dplyr::select(-factor) %>% pander(caption="Pol II samples")


# Pol II enrichment will be calculated in a window starting from the TSS and ending 200bp downstream.



enrichments_for_sample <- function(i, samples.df, txs.gr) {
  sample.bw <- file.path("../bw/", samples.df$ip[i])
  wce.bw    <- file.path("../bw/", samples.df$wce[i])

  label <- samples.df$label[i]
  factor_type <- samples.df$factor[i]
  
  regions.gr <- promoters(txs.gr, upstream=0, downstream=200)
  
  mcols(txs.gr)$region_ip  <- regionSums(regions.gr, sample.bw)
  mcols(txs.gr)$region_wce <- regionSums(regions.gr, wce.bw)

  mcols(txs.gr)$ip_total <- total_signal(sample.bw)
  mcols(txs.gr)$wce_total <- total_signal(wce.bw)

  mcols(txs.gr)$region_wce_min_coverage <- regionMins(regions.gr, wce.bw)

  mcols(txs.gr)$label <- label
  mcols(txs.gr)$factor <- factor_type

  txs.gr
}

txs.gr <- flybase_txs_granges() %>% filter_chrs()


  pol_data.df <- 1:nrow(pol_samples_df) %>%
                 mclapply(enrichments_for_sample, pol_samples_df, txs.gr, mc.cores=cores(), mc.preschedule=FALSE) %>%
                 lapply(as.data.frame) %>%
                 bind_rows 



# Before calculating enrichment, we floor the WCE signal for each region at the median WCE signal level among all transcripts.


pol_enrichments.df <- pol_data.df %>% 
                      group_by(label, factor) %>%
                      dplyr::mutate(median_region_wce = median(region_wce))

pol_enrichments.df %<>% transform(enrichment = log2((region_ip / ip_total) / (pmax(region_wce, median_region_wce) / wce_total)))

tss.infs <- which(is.infinite(pol_enrichments.df$enrichment))

min.tss <- with(pol_enrichments.df, min(enrichment[is.finite(enrichment)]))

if(length(tss.infs) > 0) pol_enrichments.df$enrichment[tss.infs] <- min.tss

pol_enrichments.df %>%
  ungroup %>%
  dplyr::select(fb_tx_id, fb_gene_id, fb_symbol, enrichment, factor, label) %>%
  saveRDS(file=figure_path("pol_enrichments.df.rds"))
pol_enrichments.df
}

if(!file.exists(figure_path("pol_enrichments.df.rds")))
   {pol_enrichments.df<-calculate_polii_enrichment()} else{
     pol_enrichments.df<-readRDS(figure_path("pol_enrichments.df.rds"))
     }
```

```{r enrichment_summary}
pol_enrichments.df %>%
  group_by(label, factor) %>%
  summarize(median_enrichment = median(enrichment),
            max_enrichment = max(enrichment)) %>%
  pander(caption="Summary of Pol II enrichments (log2)")
```

```{r divide_by_tissue_polii}

pol_single_tx.df <- pol_enrichments.df %>%
                    group_by(factor, label, fb_gene_id) %>%
                    summarise(fb_tx_id   = fb_tx_id[which.max(enrichment)],
                              enrichment = max(enrichment)) %>%
                    ungroup

pol.wide <- pol_single_tx.df %>%
            dplyr::mutate(label = paste0(factor, ".", label)) %>%
            dplyr::select(-factor, -fb_tx_id) %>%
            spread(key=label, value=enrichment)

counts.df <- pol_single_tx.df %>%
             filter(factor == "polii" & label %in% c("glia", "gut", "muscle", "neuron", "trachea", "epidermis")) %>%
             group_by(fb_gene_id) %>%
             summarize(below_bg_tissues = sum(enrichment < 0),
                       above_bg_tissues = sum(enrichment > 1))

pol.wide %<>% merge(counts.df)
```

Using a threshold of 2-fold enrichment over input, we can divide the late-induced genes by the number of tissues where Pol II is detected:

```{r late_induced_tissue_counts}
pol.wide %>% 
  filter(fb_gene_id %in% late_genes$fb_gene_id) %>%
  group_by(above_bg_tissues) %>%
  summarize(count = n()) %>%
  dplyr::select(Tissues=above_bg_tissues, everything()) %>%
  pander(caption="Late-induced genes")
```

We will also define fewer groups by combining some of the above groups:

```{r define_combined_groups}
tissue_groups <- data.frame(above_bg_tissues=0:6,
                            tissue_group4=c("0 tissues", 
                                            rep("1-2 tissues", times=2),
                                            rep("3-4 tissues", times=2),
                                            rep("5-6 tissues", times=2)))
pol.wide %<>% merge(tissue_groups)

pol.wide %>%
  filter(fb_gene_id %in% late_genes$fb_gene_id) %>%
  group_by(tissue_group4) %>%
  summarize(count = n()) %>%
  dplyr::select(Tissues=tissue_group4, everything()) %>%
  pander(caption="Alternate groups for late-induced genes")
```
## Sigalova and Bjoern groups
```{r Sigalova and Bjoern groups}

late_wide <- pol.wide %>%
             filter(fb_gene_id %in% late_genes$fb_gene_id) 


saveRDS(object =late_wide, figure_path("late_wide.df.rds"))

late_wide %>% dplyr::select(fb_gene_id) %>% write.csv(figure_path("Dataset_EV3_polii_late_genes.csv"))

rnaseq_7_groups <- split(late_wide$fb_gene_id, late_wide$above_bg_tissues)

rnaseq_4_groups <- split(late_wide$fb_gene_id, late_wide$tissue_group4)


Sigalova.df<-read.csv("data/Sigalova_2020_dataset_EV02_master_table_final_fly.csv")
Sigalova_housekeeping <- Sigalova.df %>% filter(is_housekeeping==1) %>% `$`(gene_id)
bjoern.df<-read.csv("data/bjoern_groups.csv")
Bjoern_developmental_paused <- bjoern.df %>% filter(Poised_constant_set==1) %>% `$`(flybase_gene_id)

lst_2_expanded = c(rnaseq_4_groups[c(1,4)],list(housekeeping = hk_genes$fb_gene_id,Bjoern_developmental_paused = Bjoern_developmental_paused ))

lst_4_expanded = c(rnaseq_4_groups,list(housekeeping = hk_genes$fb_gene_id,Sigalova_housekeeping= Sigalova_housekeeping,Bjoern_developmental_paused = Bjoern_developmental_paused ))

lst_7_expanded = c(rnaseq_7_groups,list(housekeeping = hk_genes$fb_gene_id,Sigalova_housekeeping= Sigalova_housekeeping,Bjoern_developmental_paused = Bjoern_developmental_paused ))

saveRDS(object =lst_2_expanded, figure_path("lst_2_expanded.lst.rds"))
saveRDS(object =lst_4_expanded, figure_path("lst_4_expanded.lst.rds"))
saveRDS(object =lst_7_expanded, figure_path("lst_7_expanded.lst.rds"))

1:7 %>% lapply(function(x){data.frame(flybase_gene_id=lst_4_expanded[[x]],group_name=names(lst_4_expanded[x]))}) %>% bind_rows() %>% filter(group_name != "Sigalova_housekeeping") %>% write.csv(figure_path("Dataset_EV4_gene_groups.csv"))

```  

## RNA-seq expression

```{r rnaseq_expression}

expression_plot <- function(groups, rnaseq) {
  exp.df <- groups %>%
            lapply(function(gene_ids) {
              subset(rnaseq, fb_gene_id %in% gene_ids)
            }) %>%
            bind_rows(.id="group_name")
  g <- ggplot(exp.df, aes(x=group_name, y=log2(wt_14to17h))) +
       geom_boxplot() + 
       theme_bw() + theme_cowplot()+
       labs(x="", y="Late embryo expression (log2 FPKM)")
  g
}

expression_plot_14and17 <- function(groups, rnaseq) {
  exp.df <- groups %>%
            lapply(function(gene_ids) {
              subset(rnaseq, fb_gene_id %in% gene_ids)
            }) %>%
            bind_rows(.id="group_name")
  g <- ggplot(exp.df, aes(x=group_name, y=log2(value), fill= group_name)) +
       geom_boxplot() +facet_grid(~variable)+ 
       theme_bw() + ylim(c(0,10))+ theme_cowplot()+
       labs(x="", y="Late expression (log2 TPM)")
  g
}
calculate_expression_pvalue <- function(groups, rnaseq, test_group_1, test_group_2, variable, alternative) {
  exp.df <- groups %>%
            lapply(function(gene_ids) {
              subset(rnaseq, fb_gene_id %in% gene_ids)
            }) %>%
            bind_rows(.id="group_name")
wilcox.test(exp.df%>% filter(group_name ==  test_group_1) %>% dplyr::select(variable) %>% as.matrix,exp.df%>% filter(group_name == test_group_2) %>% dplyr::select(variable) %>% as.matrix, alternative = alternative)
}


expression_plot(rnaseq_4_groups, rnaseq)

expression_plot(rnaseq_7_groups, rnaseq)

expression_plot(lst_2_expanded, rnaseq)

expression_plot(lst_4_expanded, rnaseq)

expression_plot(lst_7_expanded, rnaseq)


calculate_expression_pvalue(rnaseq_4_groups, rnaseq, '0 tissues','5-6 tissues','wt_14to17h','two.sided')

calculate_expression_pvalue(rnaseq_4_groups, rnaseq2, '0 tissues','5-6 tissues','wt_14to14.5h','two.sided')

calculate_expression_pvalue(rnaseq_4_groups, rnaseq2, '0 tissues','5-6 tissues','wt_17to17.5h','two.sided')

rnaseq2 %<>% gather(variable,value,-1:-9)

expression_plot_14and17(rnaseq_4_groups, rnaseq2)

expression_plot_14and17(rnaseq_7_groups, rnaseq2)

```
## Promoter element enrichments
```{r promoter_elements_by_late_housekeeping, fig.width=11, fig.height=6}

promoter_element_heatmap<-function(plot.df, 
                                     title,
                                     element.order=c("GAGA","TATA", "Inr", "DPE", "PB","DRE", "Motif1", "Motif6", "Motif7"),
                                     manual.scale=c(),
                                     star_pvalue=0.05) {

  plot.df <- transform(plot.df, enrichment = ifelse(enrichment < 1, -1 / enrichment, enrichment),
                                sig_label = ifelse(pvalue < star_pvalue, "*", ""))

  if(length(manual.scale) > 0) {
    plot.df$enrichment[plot.df$enrichment<min(manual.scale)]<-min(manual.scale)
    plot.df$enrichment[plot.df$enrichment>max(manual.scale)]<-max(manual.scale)
  } 

  plot.df <- subset(plot.df, element %in% element.order)
  plot.df$element <- factor(plot.df$element, levels=element.order)

  g <- ggplot(plot.df, aes(x=element, y=group_name, fill=enrichment)) + 
       geom_tile() + 
       geom_text(aes(label=sig_label), color="white") +
       theme_bw() +
       scale_x_discrete(expand=c(0, 0)) +
       scale_y_discrete(expand=c(0, 0)) +
       scale_fill_gradientn(name="Enrichment", space="Lab", na.value="yellow",
                            values=c(0, 0.4,0.6,1), 
                            colours=c("#000000", "#cccccc", "#cccccc", "#FC8F00"), guide="colourbar") +
       labs(x="", y="", title=title) +
       theme(panel.grid.minor=element_blank(),
             panel.grid.major=element_blank())
  g
}

select_txs <- function(gene_ids, pol.df) {
  highest_txs <- pol.df %>%
                 filter(factor == "polii" & label == "embryo14") %>%
                 filter(fb_gene_id %in% gene_ids) %>%
                 group_by(fb_gene_id) %>%
                 summarize(fb_tx_id = fb_tx_id[which.max(enrichment)])
  highest_txs$fb_tx_id
}

promoter_element_plot <- function(groups, pol.df, rnaseq) {
  pe_groups <- groups %>% lapply(select_txs, pol.df)
  names(pe_groups) <- paste0(names(pe_groups), "\n(", elementNROWS(pe_groups), ")")
  all_txs <- select_txs(rnaseq$fb_gene_id, pol.df)

  pe.df <- promoter_element_enrichments(pe_groups, all_txs)
  pe.df$group_name %<>% factor(., levels=unique(.))
  
  g <- promoter_element_heatmap(pe.df, title="Late genes",manual.scale=c(4,-4))
  list(plot=g, data=pe.df)
}

lst<-list(late_genes=late_genes$fb_gene_id,hk_genes = hk_genes$fb_gene_id)

pe_results <- lst %>% promoter_element_plot(pol_enrichments.df, rnaseq)

pe_results$plot

pe_results$data %>% dplyr::select(test_type,pvalue,enrichment,element,group_name) %>% write.csv(figure_path("Dataset_EV5_promoter_elements_late_housekeeping.csv"))
```


## Promoter element enrichments

```{r promoter_elements, fig.width=11, fig.height=6}

promoter_element_heatmap<-function(plot.df, 
                                     title,
                                     element.order=c("GAGA","TATA", "Inr", "DPE", "PB","DRE", "Motif1", "Motif6", "Motif7"),
                                     manual.scale=c(),
                                     star_pvalue=0.05) {

  plot.df <- transform(plot.df, enrichment = ifelse(enrichment < 1, -1 / enrichment, enrichment),
                                sig_label = ifelse(pvalue < star_pvalue, "*", ""))

  if(length(manual.scale) > 0) {
    plot.df$enrichment[plot.df$enrichment<min(manual.scale)]<-min(manual.scale)
    plot.df$enrichment[plot.df$enrichment>max(manual.scale)]<-max(manual.scale)
  } 

  plot.df <- subset(plot.df, element %in% element.order)
  plot.df$element <- factor(plot.df$element, levels=element.order)

  g <- ggplot(plot.df, aes(x=element, y=group_name, fill=enrichment)) + 
       geom_tile() + 
       geom_text(aes(label=sig_label), color="white",size=15) +
       theme_bw() +
       scale_x_discrete(expand=c(0, 0)) +
       scale_y_discrete(expand=c(0, 0)) +
       scale_fill_gradientn(name="Enrichment", space="Lab", na.value="yellow",
                            values=c(0, 0.4,0.6,1), 
                            colours=c("#000000", "#cccccc", "#cccccc", "#FC8F00"), guide="colourbar") +
       labs(x="", y="", title=title) +
       theme(panel.grid.minor=element_blank(),
             panel.grid.major=element_blank())
  g
}

select_txs <- function(gene_ids, pol.df) {
  highest_txs <- pol.df %>%
                 filter(factor == "polii" & label == "embryo14") %>%
                 filter(fb_gene_id %in% gene_ids) %>%
                 group_by(fb_gene_id) %>%
                 summarize(fb_tx_id = fb_tx_id[which.max(enrichment)])
  highest_txs$fb_tx_id
}

promoter_element_plot <- function(groups, pol.df, rnaseq) {
  pe_groups <- groups %>% lapply(select_txs, pol.df)
  names(pe_groups) <- paste0(names(pe_groups), "\n(", elementNROWS(pe_groups), ")")
  all_txs <- select_txs(rnaseq$fb_gene_id, pol.df)

  pe.df <- promoter_element_enrichments(pe_groups, all_txs)
  pe.df$group_name %<>% factor(., levels=unique(.))
  
  g <- promoter_element_heatmap(pe.df, title="Late genes by tissues with Pol II",manual.scale=c(4,-4))
  list(plot=g, data=pe.df)
}

{rnaseq_7_groups %>% promoter_element_plot(pol_enrichments.df, rnaseq)}$plot

{rnaseq_4_groups %>% promoter_element_plot(pol_enrichments.df, rnaseq)}$plot

{lst_2_expanded %>% promoter_element_plot(pol_enrichments.df, rnaseq)}$plot
{lst_4_expanded %>% promoter_element_plot(pol_enrichments.df, rnaseq)}$plot
lst_7_expanded %>% promoter_element_plot(pol_enrichments.df, rnaseq)->g
g$plot


lst_2_expanded %>% lapply(select_txs, pol_enrichments.df) %>% saveRDS(figure_path("lst_2_expanded_tx.lst.rds"))
lst_4_expanded %>% lapply(select_txs, pol_enrichments.df) %>% saveRDS(figure_path("lst_4_expanded_tx.lst.rds"))
lst_7_expanded %>% lapply(select_txs, pol_enrichments.df) %>% saveRDS(figure_path("lst_7_expanded_tx.lst.rds"))

g$data %>% dplyr::select(test_type,pvalue,enrichment,element,group_name) %>% filter(group_name!="Sigalova_housekeeping\n(1933)") %>% write.csv(figure_path("Dataset_EV6_promoter_elements_all_groups.csv"))
```


## ImaGO terms

```{r imago_terms}

imago_plot <- function(groups) {
  
  imago <- readRDS("data/external/insitu/insitu.rds") %>%
           filter(stage %in% c("stage11-12", "stage13-16"))
  
  all_terms <- imago$term %>% unique
  
  remove_terms <- c("maternal", "ubiq", "no staining") %>%
                  lapply(grep, all_terms) %>%
                  do.call(c, .)
  
  all_terms <- all_terms[-remove_terms]
  
  imago %<>% filter(term %in% all_terms)
  
  counts.df <- groups %>%
               lapply(function(gene_ids) {
                 imago %>% filter(fb_gene_id %in% gene_ids) %>%
                   group_by(fb_gene_id) %>%
                   summarize(unique_terms = length(unique(term)))
               }) %>%
               bind_rows(.id="group_name")
  
  medians.df <- counts.df %>%
                group_by(group_name) %>%
                summarize(genes = n(),
                          mean_count = mean(unique_terms)) %>%
                ungroup %>%
                arrange(mean_count)
                
  medians.df %>% pander(caption="Genes with ImaGO annotations")
  
  counts.df$group_name %<>% factor(levels=medians.df$group_name)
  
  g <- ggplot(counts.df, aes(x=group_name, y=unique_terms,fill = group_name)) +
       geom_boxplot() +
       theme_bw() + theme_cowplot()+
       labs(x="", y="Number of unique terms per gene")
  
  list(plot=g,data=counts.df)
}

calculate_counts_pvalue <- function(groups,test_group_1, test_group_2, variable, alternative) {
  imago <- readRDS("data/external/insitu/insitu.rds") %>%
           filter(stage %in% c("stage11-12", "stage13-16"))
  
  all_terms <- imago$term %>% unique
  
  remove_terms <- c("maternal", "ubiq", "no staining") %>%
                  lapply(grep, all_terms) %>%
                  do.call(c, .)
  
  all_terms <- all_terms[-remove_terms]
  
  imago %<>% filter(term %in% all_terms)
  
  counts.df <- groups %>%
               lapply(function(gene_ids) {
                 imago %>% filter(fb_gene_id %in% gene_ids) %>%
                   group_by(fb_gene_id) %>%
                   summarize(unique_terms = length(unique(term)))
               }) %>%
               bind_rows(.id="group_name")
  
  medians.df <- counts.df %>%
                group_by(group_name) %>%
                summarize(genes = n(),
                          mean_count = mean(unique_terms)) %>%
                ungroup %>%
                arrange(mean_count)
                
  counts.df$group_name %<>% factor(levels=medians.df$group_name)
  
wilcox.test(counts.df%>% filter(group_name ==  test_group_1) %>% dplyr::select(variable) %>% as.matrix,counts.df%>% filter(group_name == test_group_2) %>% dplyr::select(variable) %>% as.matrix, alternative = alternative)
}

{rnaseq_7_groups %>% imago_plot}$plot
rnaseq_4_groups %>% imago_plot ->g

g$plot

g$data %>% write.csv(figure_path("Dataset_EV7_imago_terms.csv"))

calculate_counts_pvalue(rnaseq_7_groups, '0','6','unique_terms','two.sided')

calculate_counts_pvalue(rnaseq_4_groups, '0 tissues','5-6 tissues','unique_terms','two.sided')

```


## Gene length

``` {r gene_length}

gene_length_plot <- function(groups) {
  txs.df <- flybase_txs_granges() %>%
            filter_chrs %>%
            as.data.frame
  
  sizes.df <- groups %>%
              lapply(function(gene_ids) {
                txs.df %>%
                  filter(fb_gene_id %in% gene_ids) %>%
                  group_by(fb_gene_id) %>%
                  summarize(width = max(width))
              }) %>%
              bind_rows(.id="group_name")
  
  g <- ggplot(sizes.df, aes(x=group_name, y=log10(width), fill =group_name)) +
       geom_boxplot() +
       theme_bw() + theme_cowplot()+
       labs(x="", y="Gene length (log10 base pairs)")
  list(plot=g,data=sizes.df) 
}

calculate_lengths_pvalue <- function(groups,test_group_1, test_group_2, variable, alternative) {
  txs.df <- flybase_txs_granges() %>%
            filter_chrs %>%
            as.data.frame
  
  sizes.df <- groups %>%
              lapply(function(gene_ids) {
                txs.df %>%
                  filter(fb_gene_id %in% gene_ids) %>%
                  group_by(fb_gene_id) %>%
                  summarize(width = max(width))
              }) %>%
              bind_rows(.id="group_name")
  
sizes.df %>% {wilcox.test(filter(. , group_name ==  test_group_1) %>% dplyr::select(variable) %>% as.matrix,filter(. , group_name == test_group_2) %>% dplyr::select(variable) %>% as.matrix, alternative = alternative)}
}

{rnaseq_7_groups %>% gene_length_plot}$plot
{rnaseq_4_groups %>% gene_length_plot}$plot

lst_2_expanded %>% gene_length_plot->g
g$plot
g$data %>% saveRDS(figure_path("gene_length.rds"))

{lst_4_expanded %>% gene_length_plot}$plot

lst_7_expanded %>% gene_length_plot -> g

g$plot
g$data %>% write.csv(figure_path("Dataset_EV8_gene_length.csv"))


calculate_lengths_pvalue(rnaseq_7_groups, '0','6','width','two.sided')

calculate_lengths_pvalue(rnaseq_4_groups, '0 tissues','5-6 tissues','width','two.sided')

```
## Go analysis tata and highly paused genes
```{r go_analysis_tata_paused, results="asis"}

go_term_analysis_tata_paused<-function(){panderOptions('knitr.auto.asis', FALSE)

gene_groups <- list("TATA genes"=rnaseq_4_groups[[1]],
                    "Paused genes"=rnaseq_4_groups[[4]])

gene_universe <- rnaseq$fb_gene_id


go_results<-gene_groups %>%
    lapply(run_go, gene_universe) %>%
    bind_rows(.id="gene_group")


for(i in unique(go_results$gene_group)) {
  filter(go_results, gene_group == i & Size >= 5) %>%
    arrange(Pvalue) %>%
    head(10) %>%
    dplyr::select(-gene_group, -OddsRatio, -ExpCount) %>%
    pander(caption=paste0("Gene group: ", i))
}

write.csv(go_results, file=figure_path("Dataset_EV9_go_results_tata_paused.csv"))


}
if(!file.exists(figure_path("Dataset_EV9_go_results_tata_paused.csv"))){go_term_analysis_tata_paused()}


#Late genes go terms figure based on excel sheet

read.csv(figure_path("Dataset_EV9_go_results_tata_paused.csv")) ->dfs

dfs %<>% filter(Term %in% c("chitin-based cuticle development","proteolysis","amino sugar metabolic process","cell wall macromolecule catabolic process","","body morphogenesis","defense response to Gram-positive bacterium","chitin-based embryonic cuticle biosynthetic process","phenol-containing compound metabolic process","generation of neurons","response to wounding","learning or memory")) 

dfs$Term %<>% factor(c("learning or memory","response to wounding","generation of neurons","phenol-containing compound metabolic process","defense response to Gram-positive bacterium","body morphogenesis","cell wall macromolecule catabolic process","amino sugar metabolic process","chitin-based embryonic cuticle biosynthetic process","proteolysis","chitin-based cuticle development"))

dfs$Pvalue %<>% {log10(.) *(-1)}

dfs %>% filter(gene_group == "TATA genes") %>% ggplot(aes(Term,Pvalue))+geom_bar(stat="identity") + coord_flip()

dfs %>% filter(gene_group == "Paused genes") %>% ggplot(aes(Term,Pvalue))+geom_bar(stat="identity") + coord_flip()

```


## ATAC 14-17h

``` {r atac_metagene, eval=TRUE, fig.width=12, fig.height=6}

tss.gr<-get(load("/data/effector_genes/temp/cage/fastq/all_mrna_cage_annotated_tss_150.RData"))


ATAC_plot <- function(groups, tss.gr) {
  rank_groups <- groups %>%
                 lapply(function(tx_id) {
                   subset(tss.gr, fb_tx_id %in% tx_id)
                 })
  names(rank_groups) %<>% gsub("\\n", " ", .)
  
  ATAC.bw <- "../bw/rpm/dme_emb_14_17h_atac_1.bam_from0to100_rpm.bw"

  
  ATAC_reads.df <- multisample_standard_metapeak(rank_groups, list("ATAC 14-17h"=ATAC.bw ), 
                                                  upstream=500, downstream=1000, smooth=15, cores=1)
  
  g <- ggplot(ATAC_reads.df, aes(x=tss_distance, y=reads, color=peak_group)) +
       geom_line() +
       scale_color_discrete("Late embryo Pol II tissue count") +
       theme_bw() +theme_cowplot()+
       labs(x="Distance to TSS", y="Average ATAC signal") 
  
  g
}

  readRDS("main_analysis/lst_7_expanded_tx.lst.rds") %>% ATAC_plot(tss.gr)  
  
   readRDS("main_analysis/lst_4_expanded_tx.lst.rds") %>% ATAC_plot(tss.gr) 
   
    readRDS("main_analysis/lst_2_expanded_tx.lst.rds") %>% ATAC_plot(tss.gr) 
    
#boxplot with the atac signal in different groups; -150 to 0bp    

atac_samples_lst<-list(atac_14h="dme_emb_14_17h_atac_1.bam_from0to100_rpm.bw")    

signal_for_samples <- function(i, samples.lst, txs.gr,window_downstream,window_upstream) {
  sample.bw <- file.path("../bw/rpm/", samples.lst[[i]])

  regions.gr <- resize(txs.gr,fix="start", width=1) 
  
  regions.gr <- resize(regions.gr, fix="start" ,width=window_downstream)
  
  regions.gr <- resize(regions.gr, fix="end" ,width=window_downstream-window_upstream)
  
  mcols(txs.gr)$start_r<-start(regions.gr)
  mcols(txs.gr)$end_r<-end(regions.gr)
  mcols(txs.gr)$region_ip  <- regionSums(regions.gr, sample.bw)
  txs.gr
}

txs.gr <- flybase_txs_granges() %>% filter_chrs()


atac_data.df <- 1:length(atac_samples_lst) %>%
                 mclapply(signal_for_samples, atac_samples_lst, txs.gr, 0,-150,mc.cores=cores(), mc.preschedule=FALSE) %>%
                 lapply(as.data.frame) %>%
                 bind_rows 


atac_signals.df <- atac_data.df 

atac_signals.df %<>% transform(signal = region_ip )
        
atac_box_plot <- function(groups, atac_signal) {
  atac.df <- groups %>%
            lapply(function(tx_ids) {
              subset(atac_signal, fb_tx_id %in% tx_ids)
            }) %>%
            bind_rows(.id="group_name")
  g <- ggplot( atac.df, aes(x=group_name, y= signal, fill=group_name)) +
       geom_boxplot(notch=FALSE,outlier.shape=NA) + 
       theme_bw() + theme_cowplot()+
       labs(x="", y="ATAC") +
    theme(panel.grid.minor=element_blank(),
      axis.text.x=element_text(angle=45, vjust=1, hjust=1, size = 16),axis.text.y=element_text(size = 16))+ylim(c(0,3000))
  list(plot=g,data=atac.df)
}

calculate_pvalue <- function(df,test_group_1, test_group_2, variable, alternative='two.sided') {

df %>% {wilcox.test(filter(. , group_name ==  test_group_1) %>% dplyr::select(variable) %>% as.matrix,filter(. , group_name == test_group_2) %>% dplyr::select(variable) %>% as.matrix, alternative = alternative)}
}
 
readRDS("main_analysis/lst_2_expanded_tx.lst.rds") %>% atac_box_plot(atac_signals.df) ->g
g$plot

readRDS("main_analysis/lst_4_expanded_tx.lst.rds") %>% atac_box_plot(atac_signals.df)->g
g$plot
#TATA and highly paused
calculate_pvalue(g$data,'0 tissues','5-6 tissues','signal')
#TATA and paused

calculate_pvalue(g$data,'0 tissues','3-4 tissues','signal')
#TATA and dual TATA

calculate_pvalue(g$data,'0 tissues','1-2 tissues','signal')

readRDS("main_analysis/lst_7_expanded_tx.lst.rds") %>% atac_box_plot(atac_signals.df)->g
g$plot

g$data %>% write.csv(figure_path("Dataset_EV10_whole_embryo_atac_data.csv"))

    
    
```

## MNase 2-4h and 14-17hrs

``` {r nucleosome_metagene, eval=TRUE, fig.width=12, fig.height=6}

tss.gr<-get(load("/data/effector_genes/temp/cage/fastq/all_mrna_cage_annotated_tss_150.RData"))

mnase_plot <- function(groups, tss.gr) {
  rank_groups <- groups %>%
                 lapply(function(tx_id) {
                   subset(tss.gr, fb_tx_id %in% tx_id)
                 })
  names(rank_groups) %<>% gsub("\\n", " ", .)
  
  mnase_14_17h.bw <- "../bw/rpm/Dme_MNase_14-17h_1.granges.rds_from100to200_rpm.bw"
  
  mnase_2_4h.bw <- "../bw/rpm/Dme_MNase_2-4h_1.granges.rds_from100to200_rpm.bw"
  
  mnase_reads.df <- multisample_standard_metapeak(rank_groups, list("Mnase 2-4h"= mnase_2_4h.bw,"Mnase 14-17h"= mnase_14_17h.bw ),
                                                  upstream=500, downstream=1000, smooth=15, cores=1)
  
  g <- ggplot(mnase_reads.df, aes(x=tss_distance, y=reads, color=peak_group)) +
       geom_line(aes(alpha = sample_name)) +facet_grid(peak_group~.,scales="free")+
       scale_color_discrete("Late embryo Pol II tissue count") +
       theme_bw() +theme_cowplot()+
       labs(x="Distance to TSS", y="Average MNase signal") 
  
  g
}

  readRDS("main_analysis/lst_7_expanded_tx.lst.rds") %>% mnase_plot(tss.gr)  
  
   readRDS("main_analysis/lst_4_expanded_tx.lst.rds") %>% mnase_plot(tss.gr) 
   
    readRDS("main_analysis/lst_2_expanded_tx.lst.rds") %>% mnase_plot(tss.gr) 


```

## Tissue specific ATAC signals
```{r ATAC_plot}
## Dividing genes by Pol II presence in different tissues
rnaseq_7_groups_txs <- rnaseq_7_groups %>% lapply(select_txs, pol_enrichments.df) 
rnaseq_4_groups_txs <- rnaseq_4_groups %>% lapply(select_txs, pol_enrichments.df) 

atac_samples_df <- read.csv("data/figure5c_tissue_atac_samples.csv") %>%
                  filter(factor == "ATAC") 

atac_samples_df %>% dplyr::select(-factor) %>% pander(caption="ATAC samples")


signal_for_samples <- function(i, samples.df, txs.gr) {
  sample.bw <- file.path("../bw/rpm/", samples.df$ip[i])

  label <- samples.df$label[i]
  factor_type <- samples.df$factor[i]

  regions.gr <- resize(txs.gr,fix="start", width=1) 
  
  regions.gr <- resize(regions.gr, fix="start" ,width=samples.df$window_downstream[i])
  
  regions.gr <- resize(regions.gr, fix="end" ,width=(samples.df$window_downstream[i]+samples.df$window_upstream[i]))
  
  mcols(txs.gr)$start_r<-start(regions.gr)
  mcols(txs.gr)$end_r<-end(regions.gr)
  mcols(txs.gr)$region_ip  <- regionSums(regions.gr, sample.bw)

  mcols(txs.gr)$label <- label
  mcols(txs.gr)$factor <- factor_type
  txs.gr
}

txs.gr <- tss.gr


atac_data.df <- 1:nrow(atac_samples_df) %>%
                 mclapply(signal_for_samples, atac_samples_df, txs.gr, mc.cores=cores(), mc.preschedule=FALSE) %>%
                 lapply(as.data.frame) %>%
                 bind_rows 


atac_signals.df <- atac_data.df 

atac_signals.df %<>% transform(signal = region_ip )


tissue_atac_plot_sorted <- function(groups,atac_signals.df) {
  txs.df <- flybase_txs_granges() %>%
            filter_chrs %>%
            as.data.frame
  
  acc.df <- groups %>%
            lapply(function(tx_ids) {
              subset(atac_signals.df, fb_tx_id %in% tx_ids)
            }) %>%
            bind_rows(.id="group_name")
  
  acc.df %>% group_by(fb_tx_id) %>% summarize(arrange_signal = median(signal)) %>% arrange(arrange_signal) -> fb_tx_id_level
  acc.df$fb_tx_id %<>% {factor(.,levels=fb_tx_id_level$fb_tx_id)}
  acc.df$label %<>% {factor(.,levels=c('epidermis','muscle','neuron','gut'))}
  
  g <- acc.df %>% {.$signal[.$signal > quantile(.$signal,0.98)]<-quantile(.$signal,0.98);.$signal[.$signal < quantile(.$signal,0.05)]<-quantile(.$signal,0.05); . } %>% ggplot(aes(label,fb_tx_id,fill=log2(signal)))+geom_tile()+facet_wrap(~group_name,scales = "free")+scale_fill_gradient2(low = "white", mid = "#ffc1c1",high = "red", midpoint = 9) +theme_cowplot()+theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1))
  g 
}

tissue_atac_plot <- function(groups,atac_signals.df) {
  txs.df <- flybase_txs_granges() %>%
            filter_chrs %>%
            as.data.frame
  
  acc.df <- groups %>%
            lapply(function(tx_ids) {
              subset(atac_signals.df, fb_tx_id %in% tx_ids)
            }) %>%
            bind_rows(.id="group_name")
  
  acc.df$label %<>% {factor(.,levels=c('epidermis','muscle','neuron','gut'))}
  
  g <- acc.df %>% {.$signal[.$signal > quantile(.$signal,0.98)]<-quantile(.$signal,0.98);.$signal[.$signal < quantile(.$signal,0.05)]<-quantile(.$signal,0.05); . } %>% ggplot(aes(label,fb_tx_id,fill=log2(signal)))+geom_tile()+facet_wrap(~group_name,scales = "free")+scale_fill_gradient2(low = "white", mid = "#ffc1c1",high = "red", midpoint = 9) +theme_cowplot()+theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1))
  g 
}

readRDS("main_analysis/lst_2_expanded_tx.lst.rds") %>% tissue_atac_plot(atac_signals.df)
readRDS("main_analysis/lst_4_expanded_tx.lst.rds") %>% tissue_atac_plot(atac_signals.df)
readRDS("main_analysis/lst_7_expanded_tx.lst.rds") %>% tissue_atac_plot(atac_signals.df)

readRDS("main_analysis/lst_2_expanded_tx.lst.rds") %>% tissue_atac_plot_sorted(atac_signals.df)
readRDS("main_analysis/lst_4_expanded_tx.lst.rds") %>% tissue_atac_plot_sorted(atac_signals.df)
readRDS("main_analysis/lst_7_expanded_tx.lst.rds") %>% tissue_atac_plot_sorted(atac_signals.df)

atac_signals.df %>% dplyr::select(fb_tx_id, fb_gene_id, label, signal) %>% merge(late_wide)  %>% filter(fb_tx_id %in% (rnaseq_7_groups_txs %>% unlist)) %>% filter(tissue_group4 == '0 tissues') %>% spread(key = label, value = signal)%>% dplyr::mutate(avg_signal = (gut+muscle+neuron+epidermis)/4) %>% dplyr::select(avg_signal) %>% as.matrix ->tata_genes_atac

atac_signals.df %>% dplyr::select(fb_tx_id, fb_gene_id, label, signal) %>% merge(late_wide)  %>% filter(fb_tx_id %in% (rnaseq_7_groups_txs %>% unlist)) %>% filter(tissue_group4 == '1-2 tissues') %>% spread(key = label, value = signal)%>% dplyr::mutate(avg_signal = (gut+muscle+neuron+epidermis)/4) %>% dplyr::select(avg_signal) %>% as.matrix ->dual_tata_genes_atac

atac_signals.df %>% dplyr::select(fb_tx_id, fb_gene_id, label, signal) %>% merge(late_wide)  %>% filter(fb_tx_id %in% (rnaseq_7_groups_txs %>% unlist)) %>% filter(tissue_group4 == '3-4 tissues') %>% spread(key = label, value = signal)%>% dplyr::mutate(avg_signal = (gut+muscle+neuron+epidermis)/4) %>% dplyr::select(avg_signal) %>% as.matrix ->paused_genes_atac

atac_signals.df %>% dplyr::select(fb_tx_id, fb_gene_id, label, signal) %>% merge(late_wide)  %>% filter(fb_tx_id %in% (rnaseq_7_groups_txs %>% unlist)) %>% filter(tissue_group4 == '5-6 tissues') %>% spread(key = label, value = signal)%>% dplyr::mutate(avg_signal = (gut+muscle+neuron+epidermis)/4) %>% dplyr::select(avg_signal) %>% as.matrix ->highly_paused_genes_atac

wilcox.test(tata_genes_atac,dual_tata_genes_atac,'two.sided')
wilcox.test(tata_genes_atac,paused_genes_atac,'two.sided')
wilcox.test(tata_genes_atac,highly_paused_genes_atac,'two.sided')


atac_signals.df %>%
  ungroup %>%
  dplyr::select(fb_tx_id, fb_gene_id, label, signal) %>%
  saveRDS(file=figure_path("atac_signals.df.rds"))

atac_signals.df %>% write.csv(figure_path("Dataset_EV11_intact_atac_data.csv"))

```
## Conservation Plots
```{r conservation_plots, dev=c("png","pdf"),fig.width=14, fig.height=10}

conservation_plot <- function(groups, conservation_at_genes) {
  conservation.df <- groups %>%
            lapply(function(gene_ids) {
              subset(conservation_at_genes, gene_id %in% gene_ids)
            }) %>%
            bind_rows(.id="group_name")
  g <- ggplot( conservation.df, aes(x=group_name, y= conserv_score, fill=group_name)) +
       geom_boxplot(notch=FALSE) + 
       theme_bw() + theme_cowplot()+
       labs(x="", y="conservation") +
    theme(panel.grid.minor=element_blank(),
      axis.text.x=element_text(angle=45, vjust=1, hjust=1, size = 16),axis.text.y=element_text(size = 16))
  list(plot=g,data=conservation.df)
}

lst_2_expanded %>% conservation_plot(Sigalova.df)->g;g$plot

lst_4_expanded %>% conservation_plot(Sigalova.df)->g;g$plot

calculate_pvalue(g$data,'0 tissues','5-6 tissues','conserv_score')

lst_7_expanded %>% conservation_plot(Sigalova.df)->g;g$plot


```
## Variation Plots based on Sigalova paper variation across different Drosphila species
```{r variation_plots, dev=c("png","pdf"),fig.width=14, fig.height=10}

variation_plot <- function(groups, variation_at_genes) {
  variation.df <- groups %>%
            lapply(function(gene_ids) {
              subset(variation_at_genes, gene_id %in% gene_ids)
            }) %>%
            bind_rows(.id="group_name")
  g <- ggplot( variation.df, aes(x=group_name, y= resid_cv, fill=group_name)) +
       geom_boxplot(notch=FALSE) + 
       theme_bw() + theme_cowplot()+
       labs(x="", y="variation") +
    theme(panel.grid.minor=element_blank(),
      axis.text.x=element_text(angle=45, vjust=1, hjust=1, size = 16),axis.text.y=element_text(size = 16))
  list(plot=g,data=variation.df)
}


lst_2_expanded %>% variation_plot(Sigalova.df) ->g
g$plot

lst_4_expanded %>% variation_plot(Sigalova.df)->g
g$plot
calculate_pvalue(g$data,'0 tissues','5-6 tissues','resid_cv')

lst_7_expanded %>% variation_plot(Sigalova.df)->g
g$plot

```
## Shape plot: Narrow vs Broad
```{r shape_plots, dev=c("png","pdf"),fig.width=14, fig.height=10}

shape_plot <- function(groups, shape_at_genes) {
  shape.df <- groups %>%
            lapply(function(gene_ids) {
              subset(shape_at_genes, gene_id %in% gene_ids)
            }) %>%
            bind_rows(.id="group_name")
  g <- ggplot(shape.df, aes(x=group_name, fill=shape)) +
       geom_bar() + 
       theme_cowplot() + 
       labs(x="", y="shape") +
    theme(panel.grid.minor=element_blank(),
      axis.text.x=element_text(angle=45, vjust=1, hjust=1, size = 16),axis.text.y=element_text(size = 16))
  g
}



lst_2_expanded %>% shape_plot(Sigalova.df)
lst_4_expanded %>% shape_plot(Sigalova.df)
lst_7_expanded %>% shape_plot(Sigalova.df)
```
```{r sequence_plots, dev=c("png","pdf"),fig.width=14, fig.height=10}

get_heatmap <- function(tss, window, name){
  seq <- getSeq(Dmelanogaster, resize(tss, window, "center"))
  seq_df <- as.character(seq) %>% lapply(., function(x)strsplit(x, "")) %>% 
    unlist(., recursive = F) %>% do.call(rbind,.) %>% as.data.frame()
  
# tss %>% resize(34,fix='end') %>% resize(fix='start',10) %>% {getSeq(Dmelanogaster,.)} %>% as.character() %>% lapply(., function(x){stringr::str_count(x, "AT")+stringr::str_count(x, "TA")}) %>% unlist(., recursive = T) %>% order -> at_order 
  
#  seq_df$id <- 1:nrow(seq_df) %>% factor(levels=at_order)
  seq_df$id <- 1:nrow(seq_df)
  seq_df_m <- reshape2::melt(seq_df, id.vars  = "id")
  
  ATGC_plot <- ggplot(seq_df_m, aes(x = variable, y = id, fill = value)) + 
    geom_raster() +
    scale_fill_manual(values = c("indianred3", "#2C699B", "black", "#FDCC4E")) +
    xlab(paste(window, "bp around TSS")) + ylab("Genes") + 
    ggtitle(name) +
    theme_cowplot() +
    theme(axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          axis.text.x = element_blank())
}


get_logo <- function(tss, window, name){
  cs2 = make_col_scheme(chars=c('A', 'T', 'C', 'G'), 
                        cols=c("indianred3", "#FDCC4E", "#2C699B", "black"))
  seq <- as.vector(getSeq(Dmelanogaster, resize(tss, window, "center")))
  ggseqlogo(seq, col_scheme=cs2) + theme_cowplot()+
    theme(axis.text.x = element_text(size = 6),
          axis.ticks.x = element_line()) +ggtitle(name)
}


readRDS("main_analysis/lst_2_expanded_tx.lst.rds") %>% lapply(function(x){x[1:350]}) -> subset_lst

1:4 %>% lapply(function(x){tss.gr[tss.gr$fb_tx_id %in% subset_lst[[x]]] %>% get_heatmap(101,subset_lst[x] %>% names)})

1:4 %>% lapply(function(x){tss[tss$fb_tx_id %in% subset_lst[[x]]] %>% get_logo( 101, subset_lst[x] %>% names)})


#get_TAnumber <- function(tss, name){ tss %>% resize(34,fix='end') %>% resize(fix='start',10) %>% {getSeq(Dmelanogaster,.)} %>% as.character() %>% lapply(., function(x){stringr::str_count(x, "AT")+stringr::str_count(x, "TA")}) %>% unlist(., recursive = T) ->tss$ta_number
#tss$groupname<-name
#  tss %>% data.frame
  
#}

#readRDS("main_analysis/lst_7_expanded_tx.lst.rds")->lst

#1:4 %>% lapply(function(x){tss.gr[tss.gr$fb_tx_id %in% lst[[x]]] %>% get_TAnumber( lst[x] %>% names)})->TA.df

#{tss.gr[tss.gr$fb_tx_id %in% lst[[2]]]} %>% resize(34,fix='end') %>% resize(fix='start',10) %>% {getSeq(Dmelanogaster,.)} %>% as.character() %>% lapply(., function(x){stringr::str_count(x, "AT")+stringr::str_count(x, "TA")}) %>% unlist(., recursive = T) %>% quantile(0.70)


#g$data %>% dplyr::select(fb_gene_id=gene_id,resid_cv,length_nt) %>% right_join(TA.df) %>% ggplot(aes(ta_number %>% as.factor,length_nt,fill=groupname))+geom_boxplot()

#loess_data.df %>% dplyr::select(gene_name=gene,var.in.ht_res) %>% right_join(TA.df) %>% mutate(ta_group = ifelse(ta_number>4,"1_high",ifelse(ta_number<2,"3_low","2_medium"))) %>% ggplot(aes(groupname,var.in.ht_res))+geom_boxplot(outlier.shape=NA)

#sc_values.df %>% dplyr::select(gene_name=gene,ng.freq.in.ot) %>% right_join(TA.df) %>% mutate(width_group = ifelse(width>10000,"1_veryhigh",ifelse(width>5000,"2_high",ifelse(width<1000,"4_low","3_medium")))) %>% ggplot(aes(width_group,ng.freq.in.ot,fill=groupname))+geom_boxplot(outlier.shape=NA)
```


```{r session_info_child, child="child_docs/session_info_child.Rmd"}
```
